src/App.js

```diff
@@ -1,96 +1,106 @@
 import {nanoid} from 'nanoid';
-import {useState, useEffect, useRef} from 'react';
+import {useState, useRef} from 'react';
 import styled from 'styled-components';
 
-import {mazeArray} from './arrays/mazeArray.js';
 import MagicForm from './components/MagicForm/MagicForm.js';
 import MagicImage from './components/MagicImage/MagicImage.js';
 import MagicList from './components/MagicList/MagicList.js';
-import executeSpell from './functionsfolder/executeSpell.js';
-import findCommandMessage from './functionsfolder/findCommandMessage.js';
-import mazeEnd from './images/mazeRooms/mazeEnd.png';
-import mazeWay from './images/mazeRooms/mazeWay.png';
+import {mazeArray} from './data/mazeArray.js';
+import executeSpell from './functions/executeSpell.js';
+import findCommandMessage from './functions/finder/findCommandMessage.js';
+import findImage from './functions/finder/findImage.js';
+import findPosition from './functions/finder/findPosition.js';
 import {CodeSpan} from './stylesheet/StyledSpans.js';
 
 export default function App() {
   const [currentPosition, setCurrentPosition] = useState(mazeArray[0]); //this holds a {object}
   const [savedOrders, setSavedOrders] = useState([]);
-  const [currentBackgroundImage, setCurrentBackgroundImage] = useState({image: mazeEnd, altText: 'you start here'});
   const [isRoomEnlighten, setIsRoomEnlighten] = useState(false);
+  const [triggerCurtain, setTriggerCurtain] = useState(false);
 
   const ref = useRef();
 
-  useEffect(() => {
-    if (currentPosition.type === 'start' || currentPosition.type === 'end') {
-      setCurrentBackgroundImage({image: mazeEnd, altText: 'the end of the way'});
-    } else {
-      setCurrentBackgroundImage({image: mazeWay, altText: 'a lonely and spooky corridor'});
-    }
-  }, [currentPosition]);
+  let currentBackgroundImage = findImage(currentPosition); //this returns an object
 
   function changePosition(path) {
-    const newPosition = mazeArray.find(mazeRoom => mazeRoom.path === path);
+    const newPosition = findPosition(path);
     setCurrentPosition(newPosition);
+    setTriggerCurtain(true);
     setIsRoomEnlighten(false); //this is NOT a toggle, changed to true only in changeEnlighten below
   }
 
+  function changeTriggerCurtain() {
+    setTriggerCurtain(false);
+  }
+
   function changeEnlighten() {
     setIsRoomEnlighten(true); //this is NOT a toggle, changed to false only in changePosition above
   }
 
-  function processingLatestSpell(input) {
-    const treatedInput = input.trim().split(' ');
-    const displayedInput = treatedInput.filter(inputUnit => inputUnit.length > 0); //this is used later aswell
-    const preparedInput = displayedInput.map(checkedInputPart => checkedInputPart.toLowerCase());
+  function processingLatestSpell(instruction) {
+    const treatedInstruction = instruction.trim().split(' ');
+    const instructionValues = treatedInstruction.filter(inputUnit => inputUnit.length > 0); //this is used later aswell
```

Comment:
Just a very very minor note. 🤓 If the  `inputUnit.length` is NOT 0,  then length is `truthy `and  you can write it like this 👇
```suggestion
    const instructionValues = treatedInstruction.filter(inputUnit => inputUnit.length); 
```